src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1372,6 +1372,15 @@ internal bool containsXmlDummyNodes()
             return this.Nodes.OfType<DummyNode>().Where(node => node.OriginalNodeContent is XmlElement).Count()> 0;
         }
 
+        /// <summary>
+        /// Returns true if the workspace contains input symbols with invalid names.
+        /// </summary>
+        /// <returns></returns>
+        internal bool containsInvalidInputSymbols()
+        {
+            return this.Nodes.OfType<Nodes.CustomNodes.Symbol>().Where(node => !node.Parameter.NameIsValid).Count() > 0;
```

Comment:
you should be able to short circuit this check with any() so that you do not need to iterate all nodes in the graph to check this, just the first matching case. Consider a graph with 10,000 nodes.

Reply:
hmm - I guess it's not such a big issue actually because you filter the list down to symbol nodes first. 👍